tree/30_generic_methods/_classes_aggregate_four.cf

```diff
@@ -0,0 +1,90 @@
+{
+  classes:
+      "first_reached"  expression => "${first_prefix}_reached";
+      "second_reached" expression => "${second_prefix}_reached";
+      "third_reached"  expression => "${third_prefix}_reached";
+      "fourth_reached" expression => "${fourth_prefix}_reached";
+
+    first_reached.second_reached.third_reached.fourth_reached::
+      "${destination_prefix}_ok"           expression => "${first_prefix}_ok.${second_prefix}_ok.${third_prefix}_ok.${fourth_prefix}_ok",
+                                           scope => "namespace";
+
+      "${destination_prefix}_kept"         expression => "${first_prefix}_kept.${second_prefix}_kept.${third_prefix}_kept.${fourth_prefix}_kept",
+                                           scope => "namespace";
+
+      "${destination_prefix}_repaired"     expression => "(${first_prefix}_repaired.${second_prefix}_ok.${third_prefix}_ok.${fourth_prefix}_ok)|
+                                                          (${first_prefix}_ok.${second_prefix}_repaired.${third_prefix}_ok.${fourth_prefix}_ok)|
+                                                          (${first_prefix}_ok.${second_prefix}_ok.${third_prefix}_repaired.${fourth_prefix}_ok)|
+                                                          (${first_prefix}_ok.${second_prefix}_ok.${third_prefix}_ok.${fourth_prefix}_repaired)",
+                                           scope => "namespace";
+
+      "${destination_prefix}_reached"      expression => "${first_prefix}_reached|${second_prefix}_reached|${third_prefix}_reached|${fourth_prefix}_reached",
+                                           scope => "namespace";
+
+      "${destination_prefix}_failed"       expression => "${first_prefix}_failed|${second_prefix}_failed|${third_prefix}_failed|${fourth_prefix}_failed",
+                                           scope => "namespace";
+
+      "${destination_prefix}_denied"       expression => "${first_prefix}_denied|${second_prefix}_denied|${third_prefix}_denied|${fourth_prefix}_denied",
+                                           scope => "namespace";
+
+      "${destination_prefix}_timeout"      expression => "${first_prefix}_timeout|${second_prefix}_timeout|${third_prefix}_timeout|${fourth_prefix}_timeout",
+                                           scope => "namespace";
+
+      "${destination_prefix}_error"        expression => "${first_prefix}_error|${second_prefix}_error|${third_prefix}_error|${fourth_prefix}_error",
+                                           scope => "namespace";
+
+      "${destination_prefix}_not_ok"       expression => "!${destination_prefix}_ok",
+                                           scope => "namespace";
+
+      "${destination_prefix}_not_kept"     expression => "!${destination_prefix}_kept",
+                                           scope => "namespace";
+
+      "${destination_prefix}_not_repaired" expression => "!${destination_prefix}_repaired",
```

Comment:
you have the predicate that there are no duplicate classes (like first_prefix_success and first_prefix_error)
this should never happen, but coding defensively should be wise